### PR TITLE
gh-105042: Disable unmatched parens syntax error in python tokenize

### PR DESCRIPTION
**Pull request by lysnikolaou**
* Issue: gh-105042

### CODE REVIEW

File: Python/Python-tokenize.c [hunk 1]
```diff
@@ -64,6 +65,12 @@ tokenizeriter_new_impl(PyTypeObject *type, const char *source,
     if (extra_tokens) {
         self->tok->tok_extra_tokens = 1;
     }
+    if (ignore_unmatched_parens) {
+        self->tok->ignore_unmatched_parens = 1;
+    }
+    if (ignore_unmatched_parens) {
```
Comment (pablogsal) [inline Python/Python-tokenize.c]: Why is this here twice?

**Comment by pablogsal** [comment]
We also need a bunch of tests that cover these, one in the `inspect` module and the other in the `tokenize` module

File: Lib/tokenize.py [hunk 2]
```diff
@@ -447,7 +447,9 @@ def tokenize(readline):
 
 def _tokenize(rl_gen, encoding):
     source = b"".join(rl_gen).decode(encoding)
-    for token in _generate_tokens_from_c_tokenizer(source, extra_tokens=True):
+    for token in _generate_tokens_from_c_tokenizer(source,
+                                                   extra_tokens=True,
+                                                   ignore_unmatched_parens=True):
```
Comment (pablogsal) [inline Lib/tokenize.py]: Hummm, maybe we should reuse `extra_tokens` here. Both flags are internal and I don't see us using one instead of the other.

File: Parser/tokenizer.c [hunk 3]
```diff
@@ -2496,41 +2497,42 @@ tok_get_normal_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct t
     case ')':
     case ']':
     case '}':
-        if (!tok->level) {
+        if (!tok->ignore_unmatched_parens && !tok->level) {
```
Comment (pablogsal) [inline Parser/tokenizer.c]: Here we are also ignoring errors inside f-string: `f-string: single '}' is not allowed`. Is that intended?

**Comment by lysnikolaou** [comment]
@pablogsal This is ready for another review.

**Comment by lysnikolaou** [comment]
@pablogsal Thanks for the reminder! Tests added in https://github.com/python/cpython/pull/105061/commits/c7481a589875bc1dc7d2be29043379f8c779b6ed.

**Review by pablogsal — APPROVED** [review]
LGTM

Fantastic work!
